tests/python/common/test_propagate.py

```diff
@@ -100,9 +101,13 @@ def test_prop_nodes_topo(idtype):
     tree.ndata["x"] = F.zeros((5, 2))
     # set all leaf nodes to be ones
     tree.nodes[[1, 3, 4]].data["x"] = F.ones((3, 2))
-    dgl.prop_nodes_topo(
-        tree, message_func=mfunc, reduce_func=rfunc, apply_node_func=None
-    )
+    # Intercepting the warning: The input graph for the user-defined edge
+    # function does not contain valid edges
```

Comment:
Just curious, why not create a graph with valid edges for testing?

Reply:
To be honest, I don't understand what's wrong with these edges:
```
    tree.add_nodes(5)
    tree.add_edges(1, 0)
    tree.add_edges(2, 0)
    tree.add_edges(3, 2)
    tree.add_edges(4, 2)
```
however this warning is generated,

Reply:
If we don't know how the warning is generated, let's avoid suppressing it.
Can you revert this and only submit the change related to tests/python/common/utils/test_filter.py.

Reply:
DONE.